File: doctorsite/doctorapp/urls.py
from django.urls import path
from rest_framework.urlpatterns import format_suffix_patterns

from doctorapp import views

app_name = 'patients'
urlpatterns = [

    path('redcap/patients/', views.RedcapPatientList.as_view(), name='redcap-api-list'),
    path('redcap/drugs/', views.RedcapDrugList.as_view(), name='redcap-api-list'),
    path('redcap/patients/<int:id>/', views.RedcapPatientDetail.as_view(), name='redcap-api-detail'),
    path('redcap/patients/<int:id>/confirm_changes', views.RedcapPatientConfirm.as_view(), name='redcap-api-confirm'),
    path('redcap/patients/<int:id>/report.pdf', views.RedcapPatientReportPdf.as_view(), name='redcap-api-report'),
    path('redcap/patients/<int:id>/poll_regimen_computed',
         views.RedcapPatientRegimenComputed.as_view(), name='redcap-api-regimen-computed'),

]

urlpatterns = format_suffix_patterns(urlpatterns)

File: doctorsite/frontend/src/utils/urls.js

export const formatPdfReportURL = patientId => `/api/redcap/patients/${patientId}/report.pdf`;
export const formatConfirmChangesURL = patientId => `/api/redcap/patients/${patientId}/confirm_changes`;
export const formatPatientLink = patientId => `/app/patient/${patientId}/`;
export const formatPatientURL = patientId => `/api/redcap/patients/${patientId}/`;
export const formatPollRegimenChangesURL = patientId => `/api/redcap/patients/${patientId}/poll_regimen_computed`;
export const homeURL = `/app`;
export const drugsURL = `/api/redcap/drugs`;
export const patientsURL = `/api/redcap/patients`;
